assignment2/q1.py

```diff
@@ -0,0 +1,36 @@
+# Print AncestorsGiven a Binary Tree and a key, 
+# write a function that prints all the ancestors of the key in the givenbinary tree.
+class Node:
+ 
+    
+    def __init__(self, data):
+        self.data = data 
+        self.left = None
+        self.right = None
+
+    def printAncestors(root, target):
+
```

Comment:
I really like your solution! There is minimum amount of code and it is very elegant! 
However, I don't really seem to be able to run it without modifications... My python interpreter tells me that **line 37, name 'printAncestors' is not defined**. It worked after the following modifications: 
```python
def printAncestors(self, root, target):
   ...
   if (self.printAncestors(...) or self.printAncestors(...)):
   ...
...
root.printAncestors(root, 5)
```
I am quite new to Python, so would really appreciate an explanation for this TAT.